libxfce4ui/xfce-gtk-extensions.c

```diff
@@ -49,6 +49,464 @@
+  gtk_menu_shell_append (menu, item);
+  gtk_widget_show (item);
+}
+
+
+
+/**
+ * xfce_gtk_accel_group_append:
+ * @accel_group : The #GtkAccelGroup to which the passed action_entries should be appended
+ * @action_entries : array of action_entries to be added
+ * @n_action_entries : size of the action_entries array
+ * @callback_data : callback data which should be passed if the accelerator is activated
+ *
+ * Connects the accel_pathes of the passed action_entries to the accel_group
+ * and adds the default key of each ActionEntry to the accel_map, if no key was defined for the related accel_path so far.
+ *
+ * Return value: (transfer none): The #GtkAccelGroup to which the passed action_entries were appended
+ *
+ * Since: 4.16
+ **/
+GtkAccelGroup*
+xfce_gtk_accel_group_append (GtkAccelGroup            *accel_group,
+                             const XfceGtkActionEntry *action_entries,
+                             guint                     n_action_entries,
+                             gpointer                  callback_data)
+{
+  GtkAccelKey key;
+  GClosure   *closure = NULL;
+
+  g_return_val_if_fail (GTK_IS_ACCEL_GROUP (accel_group), NULL);
+
+  for (size_t i = 0; i < n_action_entries; i++)
+    {
+      if (action_entries[i].accel_path == NULL || g_strcmp0 (action_entries[i].accel_path, "") == 0)
+        continue;
+
+      /* If the accel path was not loaded to the acel_map via file, we add the default key for it to the accel_map */
+      if (gtk_accel_map_lookup_entry (action_entries[i].accel_path, &key) == FALSE)
+        {
+          gtk_accelerator_parse (action_entries[i].default_accelerator, &key.accel_key, &key.accel_mods);
+          gtk_accel_map_add_entry (action_entries[i].accel_path, key.accel_key, key.accel_mods);
+        }
+      if (action_entries[i].callback != NULL)
+        {
+          closure = g_cclosure_new_swap (action_entries[i].callback, callback_data, NULL);
+          gtk_accel_group_connect_by_path (accel_group, action_entries[i].accel_path, closure);
+        }
+    }
+  return accel_group;
+}
+
+
+
+/**
+ * xfce_gtk_get_action_entry_by_id:
+ * @action_entries : array of action_entries to be searched
+ * @n_action_entries : size of the action_entries array
+ * @id : id of the action entry (usually enum values are used)
+ *
+ * Convenient method to find a specific action_entry from an array of action_entries
+ *
+ * Return value: (transfer none): The matching #XfceGtkActionEntry or NULL if not found
+ *
+ * Since: 4.16
+ **/
+const XfceGtkActionEntry*
+xfce_gtk_get_action_entry_by_id (const XfceGtkActionEntry *action_entries,
+                                 guint                     n_action_entries,
+                                 guint                     id)
+{
+  for (size_t i = 0; i <  n_action_entries; i++)
+    {
+      if( action_entries[i].id == id )
+          return &(action_entries[i]);
+    }
+  g_error ("There is no action with the id '%i'.", id);
```

Comment:
Perhaps a warning would be sufficient?

Reply:
Yes, possible better to just use g_warning
g_error will terminate the application, probably better to keep it running and return NULL

Reply:
Should still work fine together with `xfce_gtk_***_new_from_action_entry` if I add  `g_return_if_fail (action_entry != NULL)` there .. which anyhow might be a good idea.

Reply:
fixed: https://git.xfce.org/xfce/libxfce4ui/commit/?id=6742bd77c282358df7217df969420649c8fbf304

Reply:
Thanks!